// ionic-native/dist/es5/plugins/jpush.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
/**
 * This is a template for new plugin wrappers
 *
 * TODO:
 * - Add/Change information below
 * - Document usage (importing, executing main functionality)
 * - Remove any imports that you are not using
 * - Add this file to /src/index.ts (follow style of other plugins)
 * - Remove all the comments included in this template, EXCEPT the @Plugin wrapper docs.
 * - Remove this note
 *
 */
var plugin_1 = require('./plugin');
/**
 * @name JPush
 * @description
 * This plugin does something
 *
 * @usage
 * ```
 * import { JPush } from 'ionic-native';
 *
 * JPush.functionName('Hello', 123)
 *   .then((something: any) => doSomething(something))
 *   .catch((error: any) => console.log(error));
 *
 * ```
 */
var JPush = (function () {
    function JPush() {
    }
    // android && iOS
    JPush.init = function () { return; };
    // android && iOS
    JPush.stopPush = function () { return; };
    // android && iOS
    JPush.resumePush = function () { return; };
    // android && iOS
    JPush.isPushStopped = function () { return; };
    // android && iOS
    JPush.getRegistrationID = function () { return; };
    // android && iOS
    JPush.setTagsWithAlias = function (tags, alias) { return; };
    // android && iOS
    JPush.setTags = function (tags) { return; };
    // android && iOS
    JPush.setAlias = function (alias) { return; };
    // android && iOS
    JPush.getUserNotificationSettings = function () { return; };
    // iOS
    JPush.startJPushSDK = function () { return; };
    // iOS
    JPush.setBadge = function (value) { return; };
    // iOS
    JPush.resetBadge = function () { return; };
    // iOS
    JPush.setApplicationIconBadgeNumber = function (badge) { return; };
    // iOS
    JPush.getApplicationIconBadgeNumber = function () { return; };
    // iOS
    JPush.addLocalNotificationForIOS = function (delayTime, content, badge, notificationID, extras) { return; };
    // iOS
    JPush.deleteLocalNotificationWithIdentifierKeyInIOS = function (identifierKey) { return; };
    // iOS
    JPush.clearAllLocalNotifications = function () { return; };
    // iOS
    JPush.startLogPageView = function (pageName) { return; };
    // iOS
    JPush.stopLogPageView = function (pageName) { return; };
    // iOS
    JPush.beginLogPageView = function (pageName, duration) { return; };
    // iOS
    JPush.setDebugModeFromIos = function () { return; };
    // iOS
    JPush.setLogOFF = function () { return; };
    // iOS
    JPush.setCrashLogON = function () { return; };
    // iOS
    JPush.setLocation = function (latitude, longitude) { return; };
    // iOS
    JPush.isPlatformIOS = function () { return; };
    // iOS
    JPush.addDismissActions = function (actions, categoryId) { return; };
    // iOS
    JPush.addNotificationActions = function (actions, categoryId) { return; };
    // android
    JPush.setDebugMode = function (mode) { return; };
    // android
    JPush.receiveMessageInAndroidCallback = function (data) { return; };
    // android
    JPush.openNotificationInAndroidCallback = function (data) { return; };
    // android
    JPush.setStatisticsOpen = function (bool) { return; };
    // android
    JPush.reportNotificationOpened = function (msgID) { return; };
    // android
    JPush.clearAllNotification = function () { return; };
    // android
    JPush.clearNotificationById = function (notificationId) { return; };
    // android
    JPush.setPushTime = function (days, startHour, endHour) { return; };
    // android
    JPush.setSilenceTime = function (startHour, startMinute, endHour, endMinute) { return; };
    // android
    JPush.setLatestNotificationNum = function (num) { return; };
    // android
    JPush.addLocalNotification = function (builderId, content, title, notificaitonID, broadcastTime, extras) { return; };
    // android
    JPush.removeLocalNotification = function (notificationID) { return; };
    // android
    JPush.clearLocalNotifications = function () { return; };
    __decorate([
        plugin_1.Cordova()
    ], JPush, "init", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "stopPush", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "resumePush", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "isPushStopped", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "getRegistrationID", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setTagsWithAlias", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setTags", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setAlias", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "getUserNotificationSettings", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "startJPushSDK", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setBadge", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "resetBadge", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setApplicationIconBadgeNumber", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "getApplicationIconBadgeNumber", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "addLocalNotificationForIOS", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "deleteLocalNotificationWithIdentifierKeyInIOS", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "clearAllLocalNotifications", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "startLogPageView", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "stopLogPageView", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "beginLogPageView", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setDebugModeFromIos", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setLogOFF", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setCrashLogON", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setLocation", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "isPlatformIOS", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "addDismissActions", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "addNotificationActions", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setDebugMode", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "receiveMessageInAndroidCallback", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "openNotificationInAndroidCallback", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setStatisticsOpen", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "reportNotificationOpened", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "clearAllNotification", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "clearNotificationById", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setPushTime", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setSilenceTime", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "setLatestNotificationNum", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "addLocalNotification", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "removeLocalNotification", null);
    __decorate([
        plugin_1.Cordova()
    ], JPush, "clearLocalNotifications", null);
    JPush = __decorate([
        plugin_1.Plugin({
            pluginName: 'JPush',
            plugin: 'jpush-phonegap-plugin',
            pluginRef: 'plugins.jPushPlugin',
            repo: 'https://github.com/jpush/jpush-phonegap-plugin',
            platforms: ['Android', 'iOS']
        })
    ], JPush);
    return JPush;
}());
exports.JPush = JPush;
